File: solutions/0458. Poor Pigs/0458.cs
// Time complexity: O(log(buckets))
// Space complexity: O(1)
public class Solution {
    public int PoorPigs(int buckets, int minutesToDie, int minutesToTest) {
        int tests = minutesToTest / minutesToDie;
        int pigs = 0;
        
        while (Math.Pow(tests + 1, pigs) < buckets)
        {
            pigs++;
        }
        
        return pigs;
    }
}

File: solutions/0458. Poor Pigs/0458.cpp
// Time complexity: O(1)
// Space complexity: O(1)
class Solution {
public:
    int poorPigs(int buckets, int minutesToDie, int minutesToTest) {
        return ceil(log2(buckets) / log2(minutesToTest / minutesToDie + 1));
    }
};